src/test/kotlin/com/pubnub/api/endpoints/access/GrantTokenTest.kt

```diff
@@ -0,0 +1,105 @@
+import com.pubnub.api.models.consumer.access_manager.v3.PNGrantTokenResult
+import com.pubnub.api.models.server.access_manager.v3.GrantTokenResponse
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.spyk
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import retrofit2.Call
+import retrofit2.Response
+
+internal class GrantTokenTest {
+    private lateinit var pubnub: PubNub
+
+    @BeforeEach
+    internal fun setUp() {
+        MockKAnnotations.init(this)
+        val pnConfiguration = PNConfiguration(userId = UserId("myUserId")).apply {
+            subscribeKey = "something"
+            secretKey = "something"
+        }
+        pubnub = spyk(PubNub(configuration = pnConfiguration))
+    }
+
+    @MockK
+    private lateinit var grantTokenEndpointMock: GrantToken
+
+    @MockK
+    private lateinit var call: Call<GrantTokenResponse>
+
+    @MockK
+    private lateinit var responseFromServer: Response<GrantTokenResponse>
+
+    @MockK
+    private lateinit var rawResponse: okhttp3.Response
+
+    @Test
+    fun can_createGrantTokenSimple() {
+        val expectedTTL = 1337
+        val authorizedUserId = UserId("authorizedUserId")
+        val expectedToken = "token_value"
+        val grantTokenResult = PNGrantTokenResult(token = expectedToken)
+        every { pubnub.grantToken(ttl = any(), authorizedUserId = any(), spaceIds = any()) } returns grantTokenEndpointMock
+        every { grantTokenEndpointMock.sync() } returns grantTokenResult
+
+        val grantTokenEndpoint = pubnub.grantToken(
+            ttl = expectedTTL,
+            authorizedUserId = authorizedUserId,
+            spaceIds = listOf(SpaceIdGrant.name(spaceId = SpaceId("mySpaceId"), read = true, delete = true))
+        )
+        val actualGrantTokenResult: PNGrantTokenResult? = grantTokenEndpoint.sync()
+        val token = actualGrantTokenResult!!.token
+
+        assertEquals(expectedToken, token)
+    }
+
+    @Test
+    fun can_createGrantToken() {
+        val expectedTTL = 1337
+        val authorizedUserIdValue = "authorizedUserId"
+        val expectedToken = "token_value"
+        grantTokenEndpointMock = spyk(
+            GrantToken(
+                pubnub = pubnub,
+                ttl = 1111,
+                meta = null,
+                authorizedUUID = authorizedUserIdValue,
+                channels = listOf(
+                    ChannelGrant.name(name = "channel1", read = true, delete = true),
+                    ChannelGrant.pattern(pattern = "channel.*", write = true, manage = true)
+                ),
+                channelGroups = emptyList(),
+                uuids = emptyList()
+            )
+        )
+
+        every { pubnub.grantToken(ttl = any(), authorizedUserId = any(), spaceIds = any()) } returns grantTokenEndpointMock
+        every { grantTokenEndpointMock.doWork(any()) } returns call
```

Comment:
If it ok if we have a test which doesn't test `doWork` in any way? I always struggle with the value that those tests bring for the cost that we pay. When we disabling some part of our code and use mocks I'm struggling even more.

Reply:
ok